application.go

```diff
@@ -17,8 +18,13 @@ type Application interface {
 	Handlers() HandlerSet
 
 	// ForeignMessageNames returns the message names that this application
-	// uses that must be communicated to some other Dogma application.
-	ForeignMessageNames() message.NameRoles
+	// uses that must be communicated beyond the scope of the application.
+	//
+	// This includes:
+	//	- commands that are produced by this application, but consumed elsewhere
+	//	- commands that are consumed by this application, but produced elsewhere
+	//	- events that are consumed by this application, but produced elsewhere
```

Comment:
Should there be another event thing here?
> `- events that are produced by this application, but consumed elsewhere`

Reply:
And so on for the others like this below.

Reply:
No, because events don't *have* to have a handler at all, so there's no requirement that an engine work out where to send those events.

Reply:
Or, put another way... it's up to the consumer to work out where to find foreign events.

Reply:
Or, put another way, it's up to the consumer to find the events it needs. Whereas an app that produces a command must actively find where to put it.